commands/project/deploy.ts

```diff
@@ -1,50 +1,60 @@
-const { pollDeployStatus } = require('../../lib/projects/buildAndDeploy');
-const { getProjectDetailUrl } = require('../../lib/projects/urls');
-const { projectNamePrompt } = require('../../lib/prompts/projectNamePrompt');
-const { promptUser } = require('../../lib/prompts/promptUtils');
-const { i18n } = require('../../lib/lang');
-const { uiBetaTag, uiLink } = require('../../lib/ui');
-const { getAccountConfig } = require('@hubspot/local-dev-lib/config');
-const { EXIT_CODES } = require('../../lib/enums/exitCodes');
-const { uiCommandReference, uiAccountDescription } = require('../../lib/ui');
-const { isHubSpotHttpError } = require('@hubspot/local-dev-lib/errors/index');
+} from '@hubspot/local-dev-lib/api/projects';
+import { getAccountConfig } from '@hubspot/local-dev-lib/config';
+import { logger } from '@hubspot/local-dev-lib/logger';
+import { isHubSpotHttpError } from '@hubspot/local-dev-lib/errors/index';
+import { useV3Api } from '../../lib/projects/buildAndDeploy';
+import { trackCommandUsage } from '../../lib/usageTracking';
+import { logError, ApiErrorContext } from '../../lib/errorHandlers/index';
+import { getProjectConfig } from '../../lib/projects';
+import { pollDeployStatus } from '../../lib/projects/buildAndDeploy';
+import { getProjectDetailUrl } from '../../lib/projects/urls';
+import { projectNamePrompt } from '../../lib/prompts/projectNamePrompt';
+import { promptUser } from '../../lib/prompts/promptUtils';
+import { i18n } from '../../lib/lang';
+import { uiBetaTag, uiLink } from '../../lib/ui';
+import { EXIT_CODES } from '../../lib/enums/exitCodes';
+import { uiCommandReference, uiAccountDescription } from '../../lib/ui';
+import {
+  CommonArgs,
+  ConfigArgs,
+  AccountArgs,
+  EnvironmentArgs,
+} from '../../types/Yargs';
+import { makeYargsBuilder } from '../../lib/yargsUtils';
 
 const i18nKey = 'commands.project.subcommands.deploy';
 
-exports.command = 'deploy';
-exports.describe = uiBetaTag(i18n(`${i18nKey}.describe`), false);
-
-const validateBuildId = (
-  buildId,
-  deployedBuildId,
-  latestBuildId,
-  projectName,
-  accountId
-) => {
+export const command = 'deploy';
+export const describe = uiBetaTag(i18n(`${i18nKey}.describe`), false);
+
+export type ProjectDeployArgs = CommonArgs &
+  ConfigArgs &
+  AccountArgs &
+  EnvironmentArgs & {
+    project?: string;
+    build?: number;
+    buildId?: number;
+  };
+
+function validateBuildId(
+  buildId: number,
+  deployedBuildId: number | undefined,
+  latestBuildId: number,
+  projectName: string | undefined,
+  accountId: number
+): boolean | string {
   if (Number(buildId) > latestBuildId) {
     return i18n(`${i18nKey}.errors.buildIdDoesNotExist`, {
       buildId: buildId,
-      projectName,
+      projectName: projectName!,
       linkToProject: uiLink(
         i18n(`${i18nKey}.errors.viewProjectsBuilds`),
-        getProjectDetailUrl(projectName, accountId)
+        getProjectDetailUrl(projectName!, accountId)!
```

Comment:
I think the updates you made to the `projectNamePrompt` in your other PR should let you get rid of a lot of these non-null assertions. `validateBuildId` could _probably_ just accept a string for `projectName`